edb/server/protocol/binary.pyx

```diff
@@ -469,6 +469,11 @@ cdef class EdgeConnection(frontend.FrontendConnection):
                 f'message: "database"'
             )
 
+        # token in the HTTP header has higher priority than the ClientHandshake
+        # message, under the binary-protocol-over-HTTP scenario
+        if self._auth_data is None:
+            self._auth_data = params.get('token')
+
```

Comment:
Change to:

```suggestion
        token = self._auth_data
        if token is None:
            token = params.get('token')
```

and move under `authmethod_name == 'JWT':` to avoid clobbering `_auth_data` for cases when auth is not JWT